server/controllers/clubProfileController.js

```diff
@@ -1,6 +1,54 @@
-const clubOverview = require("../models/ClubOverviewModel")
 const clubProfile = require("../models/ClubProfileModel")
 
+const findSimilarClubs = (res, clubResult) => {
+    clubProfile.aggregate([
+      {
+        '$match': { '$and': [
+          {
+            'tags': {
+              '$in': clubResult.tags
+            }
+          },
+          {
+            '_id': { '$ne': clubResult._id }
+          }
+        ]
+        }
+      }, {
+        '$addFields': {
+          'intersection': {
+            '$setIntersection': [
+              '$tags', clubResult.tags
+            ]
+          }
+        }
+      }, {
+        '$addFields': {
+          'length': {
+            '$size': '$intersection'
+          }
+        }
+      }, {
+        '$sort': {
+          'length': -1
+        }
+      }, {
+        '$limit': 6
```

Comment:
Would there be a way to only consider clubs with _at least one overlapping tag_ with the target?